go/vt/vtgate/engine/trace.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"context"
+	"encoding/json"
+
+	"vitess.io/vitess/go/mysql/collations"
+
+	"vitess.io/vitess/go/sqltypes"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+)
+
+var _ Primitive = (*Trace)(nil)
+
+type Trace struct {
+	identifiablePrimitive
+	Inner Primitive
+}
```

Comment:
nit: should this be made part of vexplain primitive? execute the code based on the variant